src/ui/util.rs

```diff
@@ -61,7 +62,9 @@ pub fn millis_to_minutes(millis: u128) -> String {
 pub fn display_track_progress(progress: u128, track_duration: u32) -> String {
     let duration = millis_to_minutes(u128::from(track_duration));
     let progress_display = millis_to_minutes(progress);
-    let remaining = millis_to_minutes(u128::from(track_duration) - progress);
+
+    let safe_remaining_ms = max(i128::from(track_duration) - progress as i128, 0) as u128;
```

Comment:
To make this "more safe" you could use `track_duration.checked_sub(progress)` or something to that effect?